ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BackendQuery.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.query;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class BackendQuery {
+  private static final Logger LOG = LogManager.getLogger();
+
+  public static <T> T runIfAlive(final Callable<T> task, final Supplier<Boolean> alive)
+      throws Exception {
+    return runIfAlive(Optional.empty(), task, alive);
+  }
+
+  public static <T> T runIfAlive(
+      final String taskName, final Callable<T> task, final Supplier<Boolean> alive)
+      throws Exception {
+    return runIfAlive(Optional.ofNullable(taskName), task, alive);
+  }
+
+  public static <T> T runIfAlive(
+      final Optional<String> taskName, final Callable<T> task, final Supplier<Boolean> alive)
+      throws Exception {
+    if (!alive.get()) {
+      LOG.warn(
+          "Zombie backend query detected [ {} ], aborting process.", taskName.orElse("unnamed"));
+      throw new RuntimeException("Timeout expired");
```

Comment:
Not a fan of using an exception when we can just check a boolean and exit the method.  The real question is what does the resulting JSON response look like when we throw?  If it looks like what we want the users to see then an exception will read better than a check and return error code segment in the calling RPC method.

Reply:
Well. I don't think we want to add info in the JSON response, since the HTTP timeout will be triggered first and generate the response with a `504 Gateway Timeout` error.
However i think using exceptions there make the integration smoother. Because basically you can wrap the wall in a lambda and it is pretty transparent. If you use boolean to check the validity of a backend query, you will add a lot of boilerplate code. For example, the `matchingLogs` methods triggers many backend queries and it would add numerous `if` blocks if we don't use exceptions.

Reply:
As discussed this PR will focus on JSON RPC and WebSocket. GraphQL support will come in a separate PR. Resolving discussion.